### ISSUE TITLE
Add CSVLogger to train.py

### ISSUE THREAD

**Issue by adamjstewart** [microsoft/torchgeo#705]
### Summary

Our current `train.py` script only logs to TensorBoard. We should add a CSV logger.

### Rationale

In order to determine the optimal hyperparams in a hyperparam sweep, we need an easy way of finding the final val accuracy of each run. Currently, the only way to get this is to convert the TensorBoard logger output to a CSV. If we log directly to CSV this becomes much easier, especially for those who don't use TensorBoard.
